SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java

```diff
@@ -215,6 +238,22 @@ public SingularityDeployHistory getDeploy(@ApiParam("Request ID for deploy") @Pa
   }
 
   @GET
+  @Path("/request/{requestId}/deploys/count")
+  @ApiOperation("Get deploy history with metadata for a single request")
+  public SingularityPaginatedResponse<SingularityDeployHistory> getDeploysCount(
+          @ApiParam("Request ID to look up") @PathParam("requestId") String requestId,
+          @ApiParam("Maximum number of items to return") @QueryParam("count") Integer count,
+          @ApiParam("Which page of items to view") @QueryParam("page") Integer page) {
+    authorizationHelper.checkForAuthorizationByRequestId(requestId, user, SingularityAuthorizationScope.READ);
+
+    final int dataCount = deployHistoryHelper.getBlendedHistoryCount(requestId).get();
```

Comment:
Maybe the count should just be optional on the response object rather than risking the NPE. Would be better for the ui to just handle rendering without a count than to have a bogus count or get an error probably